packages/create-sitecore-jss/src/templates/nextjs/scripts/watch-component-factory-creator.ts

```diff
@@ -0,0 +1,76 @@
+import { getProjectList, writeComponentFactoryCreator, constants } from '@sitecore-jss/sitecore-jss-dev-tools/nextjs';
+import { watchItems } from '@sitecore-jss/sitecore-jss-dev-tools';
+
+/*
+  COMPONENT FACTORY CREATOR GENERATION
+  Generates:
+    * `/src/temp/componentFactoryCreator.ts` file, which provides creator to generate
+      a componentFactory/moduleFactory based on custom configuration (e.g. editing mode)
+    * `/src/temp/projects/<project>.ts` files, that provide components implemented for each project
+      and used by componentFactoryCreator
+
+  componentFactory maps JSS React components to Sitecore renderings, while moduleFactory maps component files.
+
+  The component factory is a mapping between a string name and a React component instance.
+  When the Sitecore Layout service returns a layout definition, it returns named components.
+  This mapping is used to construct the component hierarchy for the layout.
+
+  Generating the componentFactoryCreator is optional, and it can be maintained manually if preferred.
+
+  The default convention uses the component's filename (without the extension) as the component
+  name. For example, the file `/components/ComponentName.ts` would map to component `ComponentName`.
+  This can be customized in writeComponentFactoryCreator().
+*/ 
+/*
+  This script supports utilizes factory creator in watch mode.
+  In watch mode, the component factory creator source folder is watched, and componentFactoryCreator.ts is
+  regenerated whenever files are added or deleted.
+
+  For manual component factory generation run bootstrap
+*/
+
+watchComponentFactoryCreator();
+
+/**
+ * You can specify components which you want to import from external/internal packages
+ * in format:
+ * let customPackages = [{
```

Comment:
Can we add a `const packages = []` variable in the function, essentially the same like it was before, and above the varaible you can add description.
Like it was here: https://github.com/Sitecore/jss/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/scripts/generate-component-factory.ts#L58-L71
In this case you don't need to copy paste writeComponentFactoryCreator example and the interface

Reply:
I did a one better - see https://github.com/Sitecore/jss/pull/1471/commits/7ba29f105e2d945a869fba0a49a5f0708bbb33cf
Single source of truth for paths and packages, and a stepping stone to plugin approach.
Eventually we'll probably want to invoke component factory creator through our plugin approach - but there are a lot of considerations with it, so I'll create a follow up pbi